### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by devendra-alt**
# [OOP school library: add basic UI](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_ui.md)
## 👨‍💻  requirement covered 
✔️ Added basic `UI`
✔️ Added application loop
✔️ Performed given operations
✔️ Listed different elements of the school library system
✔️ Created `main` class

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -1,14 +1,118 @@
...
+      puts '2 - List all people'
+      puts '3 - Create a person'
+      puts '4 - Create a book'
+      puts '5 - Create a rental'
+      puts '6 - List all rentals for a given person id'
+      puts '7 - Exit'
+      user_input = gets.chomp.strip.to_i
+      case_method(user_input)
+    end
+  end
```
Comment (skyv26) [inline main.rb]: - [x] Please 🙏🏻 adhere to the project requirement, according to which you should implement the `console` menu inside the `main.rb` and the rest of the logic, methods etc should be inside the `app.rb` file, but it seems like you did everything inside the `main.rb` file, which is against the requirement of the project.

image

File: main.rb [hunk 2]
```diff
@@ -1,14 +1,118 @@
...
+    print 'ID of person'
+    gets.chomp.to_i
+    choose_person = @people.each do |person|
+      return person if person.id == id
+    end
+    puts 'Rentals: '
+    p choose_person
+    return unless choose_person.length.positive?
+
+    choose_person.rentals.each do |rental|
```
Comment (skyv26) [inline main.rb]: - [x] Kindly make sure that you project functionalities are working before asking for the review, I tried to view the `6th` option `[List the rentals of the person]`, but after entering the ID I got an error related to *undefined local variable*. Please fix the errors in your code and submit the project with correct functionality.

Screenshot from 2023-09-23 20-17-30

File: main.rb [hunk 3]
```diff
@@ -1,14 +1,118 @@
...
+    book_choice = gets.chomp.to_i
+    book = @books[book_choice]
+    puts 'Select a person from the following list by number [not id]'
+    list_people
+    person_choice = gets.chomp.to_i
+    person = @people[person_choice]
+    print 'Enter date of booking [yyyy/mm/dd] : '
+    date = gets.chomp.strip
+    person.add_rental(date, book)
+  end
```
Comment (skyv26) [inline main.rb]: - [x] As I mentioned above, kindly make sure that your project functionalities are working before asking for the review, I tried to view the `5th` option `[Create the rental]`, but after entering the required data, I got an error related to *workspace/oop-school-library/school/person.rb:24:in `add_rental': uninitialized constant Person::Rental (NameError)*. Please fix the errors in your code and submit the project with the correct functionality.

Screenshot from 2023-09-23 20-26-48

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @devendra-alt , 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Implemented the features ✅ 
- Nice *Console* menu ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb
Comment (Alusp) [inline app.rb:1]: - [x] You have done good so far, :clap: , however, as requested by the previous reviewer, Please ensure that you test everything before submitting for your next review, as it is right now, your student name does not display on the console. Please resolve it. :slightly_smiling_face: 

Below is a screenshot of the console: :arrow_down: 
StudentName

**Review by Alusp — CHANGES_REQUESTED** [review]
Hi @devendra-alt, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
close

 ## Highlights

- [x] Linters set-up ✔️
- [x] Commit messages descrptive ✔️
- [x] Professional Github Flow used ✔️
- [x] Professional PR Summary Flow used ✔️
- [x] UI Closer to the given design :heavy_check_mark:  

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alusp in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb
Comment (rloterh) [reply app.rb:1]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Devendra,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing improvements to the console app!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**